client/js/utils/ajax.js

```diff
@@ -2,10 +2,12 @@ import request from 'request';
 import when from 'when';
 
 
+// FIXME: replace with XMLHTTPRequest
```

Comment:
32kb is big for what we need done.

I like the extra stuff provided by `when`, like `tap`. I was thinking of using https://github.com/cscott/prfun, since we already have `Promise` provided by ES6.